pkg/devfile/adapters/docker/component/adapter.go

```diff
@@ -259,3 +260,28 @@ func (a Adapter) Delete(labels map[string]string) error {
 	return nil
 
 }
+
+// Exec executes a command in the component
+func (a Adapter) Exec(command []string) error {
+	if !utils.ComponentExists(a.Client, a.ComponentName) {
+		return errors.Errorf("the component %s doesn't exist on the cluster", a.ComponentName)
+	}
+
+	containers, err := utils.GetComponentContainers(a.Client, a.ComponentName)
+	if err != nil {
+		return errors.Wrapf(err, "error while retrieving container for odo component %s", a.ComponentName)
+	}
+
+	runCommand, err := common.GetRunCommand(a.Devfile.Data, "")
```

Comment:
How we will handle here, if `runCommand` was passed as a `odo flag --run-command` during `odo push`?
Same problem i am getting for `odo log`

Reply:
We can add a flag for that. But during the discussion, we decided to go with the default container and command for now and implement that later.

Reply:
Any update on this?

Reply:
> But during the discussion, we decided to go with the default container and command for now and implement that later.

We had decided to use the default run container during the discussion.